sinks/signalfx/signalfx.go

```diff
@@ -208,69 +210,36 @@ func (sfx *SignalFxSink) Flush(ctx context.Context, interMetrics []samplers.Inte
 	return err
 }
 
-// FlushEventsChecks sends events to SignalFx. It does not support checks.
+var successSpanTags = map[string]string{"sink": "signalfx", "results": "success"}
+var failureSpanTags = map[string]string{"sink": "signalfx", "results": "failure"}
+
+// FlushOtherSamples sends events to SignalFx. It does not support checks.
 func (sfx *SignalFxSink) FlushOtherSamples(ctx context.Context, samples []ssf.SSFSample) {
 	span, _ := trace.StartSpanFromContext(ctx, "")
 	defer span.ClientFinish(sfx.traceClient)
-
 	for _, sample := range samples {
-
-		if _, ok := sample.Tags[dogstatsd.EventIdentifierKey]; !ok {
-			// This isn't an event, just continue
-			continue
-		}
-
-		dims := map[string]string{}
-
-		// Copy common dimensions in
-		for k, v := range sfx.commonDimensions {
-			dims[k] = v
-		}
-		// And hostname
-		dims[sfx.hostnameTag] = sfx.hostname
-
-		for k, v := range sample.Tags {
-			if k == dogstatsd.EventIdentifierKey {
-				// Don't copy this tag
-				continue
+		switch ddOtherSampleKind(sample.Tags) {
+		case ddSampleEvent:
+			ev := sfx.reportEvent(&sample)
+			// TODO: Split events out the same way as points
+			sfx.defaultClient.AddEvents(ctx, []*event.Event{&ev})
+			// report evt
```

Comment:
feels like you could add the event to the client in `reportEvent` directly, too - especially as `convertServiceCheck` does this.

Reply:
Do you mean don't return the `ev`? I chose this method because the `sfx.reportEvent` function doesn't require a `ctx` argument. I'm okay with changing it however for the sake of consistency but my intention was to make it so that it didn't have extra arguments/dependencies to mock out when testing.

Reply:
If such a test were to be written in the future.

Reply:
I think passing in the context to report would be reasonable, especially given the method name - you can still make a ctx in tests with `context.TODO()`, and I belieeeve we do some faking so we can exercise the defaultClient in tests?